dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/PoolingLayer.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2016 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.layers;
+
+import edu.snu.dolphin.dnn.blas.Matrix;
+import edu.snu.dolphin.dnn.conf.LayerConfigurationParameters.*;
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+
+/**
+ * Pooling layer.
+ *
+ * This layer is not learnable.
+ * This layer resizes input matrix spatially, using max pooling or average pooling.
+ * This layer works for only 1D and 2D inputs.
+ * In a forward pass,
+ * max pooling picks the maximum value in certain range (kernelHeight * kernelWidth) and these values make up output.
+ * Average pooling gets the average of values in certain range (kernelHeight * kernelWidth)
+ * and these values make up output.
+ * In a backward pass,
+ * each value of error matrix is the sum of elements in next error matrix on which had an impact in feedforward step.
+ */
+public final class PoolingLayer extends LayerBase {
+
+  private enum PoolType {
+    AVERAGE, MAX
+  }
+  private final int[] outputShape;
+  private PoolType poolingType;
```

Comment:
I'll fix this right away. I thought I fixed this, but I had some trouble handling git so the change was not saved. I'm sorry for bother you.